app/camera2xx/src/main/cpp/camera2xx.cpp

```diff
@@ -0,0 +1,26 @@
+#include <jni.h>
+#include <string>
+
+#include <camera/NdkCameraManager.h>
+
+extern "C"
+JNIEXPORT void JNICALL
+Java_com_kegarlv_camera2xx_NativeLib_getCameraIdList(JNIEnv *env, jobject thiz, jobject callback) {
+    std::shared_ptr<ACameraManager> cameraManager(ACameraManager_create(),
+                                                  [](ACameraManager *ptr) { ACameraManager_delete(ptr); });
+
+    //NOTE find a way to use smart pointer with cameraIdList
+    ACameraIdList *cameraIdList = nullptr;
+    ACameraManager_getCameraIdList(cameraManager.get(), &cameraIdList);
+
```

Comment:
you can init share_ptr with raw pointer, or implement factory method to minimize risk of raw pointer usage.